solidity/contracts/OngoingAirdrops.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity >=0.8.7 <0.9.0;
+
+import './utils/Governable.sol';
+import '../interfaces/IOngoingAirdrops.sol';
+import '@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol';
+import '@openzeppelin/contracts/utils/cryptography/MerkleProof.sol';
+
+abstract contract OngoingAirdrops is Governable, IOngoingAirdrops {
+  using SafeERC20 for IERC20;
+
+  /// @inheritdoc IOngoingAirdrops
+  mapping(bytes32 => bytes32) public roots;
+  /// @inheritdoc IOngoingAirdrops
+  mapping(bytes32 => uint32) public deadline;
+  /// @inheritdoc IOngoingAirdrops
+  mapping(bytes32 => uint256) public amountClaimedByCampaignTokenAndUser;
+  /// @inheritdoc IOngoingAirdrops
+  mapping(bytes32 => uint256) public totalAirdropedByCampaignAndToken;
+  /// @inheritdoc IOngoingAirdrops
+  mapping(bytes32 => uint256) public totalClaimedByCampaignAndToken;
+
+  constructor(address _governor) Governable(_governor) {}
```

Comment:
I think we should't use Governable, and should use AccessControl instead. We might want to automate this, and give access to some bots/keepers. Governable would make it much more difficult to do so

Reply:
To be honest, I don't think we are going to automate this. We are going to be sending funds automagically somewhere? I'd much rather have the bot create the transaction on our multisig, check the token transfers amounts (to check if we f*-ed up?). Having said that, I'll add access control just in case :).

Reply:
I have removed all constructor logic and control access will be implemented on #10